plugins/woocommerce/src/Admin/API/Reports/Products/DataStore.php

```diff
@@ -455,15 +551,15 @@ public static function sync_order_products( $order_id ) {
 				$tax_amount += isset( $tax_data['total'][ $tax_item_id ] ) ? (float) $tax_data['total'][ $tax_item_id ] : 0;
 			}
 
-			$net_revenue = round( $order_item->get_total( 'edit' ), $decimals );
 			if ( $round_tax ) {
 				$tax_amount = round( $tax_amount, $decimals );
 			}
 
 			$result = $wpdb->replace(
 				self::get_db_table_name(),
 				array(
-					'order_item_id'         => $order_item_id,
+					// When the order is full refund, there is no order items, so the order_item_id should be 0.
+					'order_item_id'         => $is_full_refund ? 0 : $order_item_id,
 					'order_id'              => $order->get_id(),
```

Comment:
> Since that will still make the order item ID unique in combination with the order number, why can't we just always populate it?

You're absolutely right. Not sure how did I get that far expanding the primary key to `product_id`. 😓 Updated in 7c85581.